apps/frontend/src/app/5_pages/MarketMakingPage/components/AdjustAndDepositModal/components/NewPoolStatistics/NewPoolStatistics.tsx

```diff
@@ -1,91 +1,146 @@
 import React, { FC, useMemo } from 'react';
 
+import classNames from 'classnames';
 import { t } from 'i18next';
 
 import { SupportedTokens } from '@sovryn/contracts';
 import { SimpleTable, SimpleTableRow } from '@sovryn/ui';
 import { Decimal } from '@sovryn/utils';
 
 import { AmountRenderer } from '../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import {
+  BITCOIN,
+  BTC_RENDER_PRECISION,
+  TOKEN_RENDER_PRECISION,
+} from '../../../../../../../constants/currencies';
+import { useGetRBTCPrice } from '../../../../../../../hooks/zero/useGetRBTCPrice';
 import { translations } from '../../../../../../../locales/i18n';
 import { decimalic } from '../../../../../../../utils/math';
+import { useGetAccumulatedReward } from '../../../../hooks/useGetAccumulatedReward';
+import { useGetExpectedTokenAmount } from '../../../../hooks/useGetExpectedTokenAmount';
+import { useGetUserInfo } from '../../../../hooks/useGetUserInfo';
 import { AmmLiquidityPool } from '../../../../utils/AmmLiquidityPool';
 import { AdjustType } from '../../AdjustAndDepositModal.types';
-
-// TODO: Fetch rewards in the component. If you want to test the functionality, set it to Decimal.ONE
-const transferRewards = Decimal.ZERO; // Decimal.ONE
-
-// TODO: Fetch weekly rewards estimation in the component. If you want to test the functionality, set it to Decimal.ONE
-const weeklyRewardsEstimation = Decimal.ONE; // Decimal.ONE
-
-// TODO: Fetch initialPoolBalance in the component.
-const initialPoolBalance = decimalic(20);
+import { useGetPoolBalanceAndRewards } from '../../hooks/useGetPoolBalanceAndRewards';
 
 const pageTranslations =
   translations.marketMakingPage.adjustAndDepositModal.newPoolStatistics;
 
 type NewPoolStatisticsProps = {
-  amount: Decimal;
+  value: string;
+  decimalAmount: Decimal;
   isInitialDeposit: boolean;
   adjustType: AdjustType;
   pool: AmmLiquidityPool;
 };
 
 export const NewPoolStatistics: FC<NewPoolStatisticsProps> = ({
-  amount,
+  value,
+  decimalAmount,
   isInitialDeposit,
   adjustType,
   pool,
 }) => {
-  const token = useMemo(() => pool.assetA, [pool.assetA]);
+  const tokenA = useMemo(() => pool.assetA, [pool.assetA]);
+  const { balanceA, balanceB } = useGetUserInfo(pool);
+  const { price: rbtcPrice } = useGetRBTCPrice();
+  const { reward } = useGetAccumulatedReward(pool.poolTokenA);
+  const { reward: accumulatedRewards } = useGetUserInfo(pool);
+
+  const renderRewards = useMemo(
+    () => reward.add(accumulatedRewards),
+    [reward, accumulatedRewards],
+  );
+
+  const { amount: expectedTokenAmount } = useGetExpectedTokenAmount(
+    pool,
+    decimalic(value),
```

Comment:
You use `decimalic(value)` in multiple places, create a const (e.g. `decimalValue`).